contracts/src/TeleporterTokenSource.sol

```diff
@@ -40,9 +40,8 @@ abstract contract TeleporterTokenSource is ITeleporterTokenBridge, TeleporterOwn
             => mapping(address destinationBridgeAddress => uint256 balance)
     ) public bridgedBalances;
 
-    // TODO: these are values brought from the example ERC20Bridge contract.
-    // Need to figure out appropriate values.
-    uint256 public constant SEND_TOKENS_REQUIRED_GAS = 300_000;
+    /// @notice Required gas limit for sending tokens to another chain.
+    uint256 public constant SEND_TOKENS_REQUIRED_GAS = 80_000;
```

Comment:
Given that token source contracts are intended to be compatible with any destination contracts that implement the required interface, I think we should have the gas limit for messages sent from source->destination be provided as a parameter for each call to `send`. A destination contracts can have arbitrarily complex logic, so we shouldn't try to fit one gas limit to all of them.

It could make sense to have some minimum or default that we don't allow the passed value to be less than, but I think we'll need to allow for callers to specify higher values if needed.

Reply:
added `requiredGasLimit` to `SendTokensInput` since destination token bridges also need to pass in case for multihop.

Reply:
The required gas limit for message from destination token bridge -> source is still a constant in `TeleporterTokenDestination` though

Reply:
I think the same logic applies to messages to the source, since the source can be composed with an arbitrary ERC20, which can also execute arbitrary logic.

Reply:
Good point, maybe we should have the value set in the constructor of `TeleporterTokenDestination`? Its value should be more or less constant for a given source contract, but it could change from one source to another as Cam called out.